packages/core-mobile/app/services/wallet/WalletService.tsx

```diff
@@ -649,10 +674,114 @@ class WalletService {
+      isDevnet(network)
     )
     return wallet.getReadOnlyAvaSigner({ accountIndex, provXP })
   }
+
+  public async createDummyImportPTx({
+    stakingAmount,
+    accountIndex,
+    avaxXPNetwork,
+    sourceChain,
+    destinationAddress,
+    feeState
+  }: CreateImportPTxParams & { stakingAmount: bigint }): Promise<UnsignedTx> {
+    const readOnlySigner = await this.getReadOnlyAvaSigner(
+      accountIndex,
+      avaxXPNetwork
+    )
+    const assetId = isDevnet(avaxXPNetwork)
+      ? DEVNET_AVAX_ASSET_ID
+      : avaxXPNetwork.isTestnet
+      ? TESTNET_AVAX_ASSET_ID
+      : MAINNET_AVAX_ASSET_ID
+
+    const utxos = getTransferOutputUtxos(
+      stakingAmount,
+      assetId,
+      destinationAddress ?? '',
+      DUMMY_UTXO_ID
+    )
+    const utxoSet = new utils.UtxoSet([utxos])
+    return readOnlySigner.importP({
+      utxoSet,
+      sourceChain,
+      toAddress: destinationAddress,
+      feeState
+    })
+  }
+
+  public async createDummyExportPTx({
+    amountInNAvax,
+    accountIndex,
+    avaxXPNetwork,
+    destinationChain,
+    destinationAddress,
+    feeState
+  }: CreateExportPTxParams): Promise<UnsignedTx> {
+    const readOnlySigner = await this.getReadOnlyAvaSigner(
+      accountIndex,
+      avaxXPNetwork
+    )
+    const provider = await NetworkService.getAvalancheProviderXP(
+      !!avaxXPNetwork.isTestnet,
+      isDevnet(avaxXPNetwork)
+    )
+    const assetId = provider.getAvaxID()
+    const utxos = getTransferOutputUtxos(
+      amountInNAvax * 2n,
+      assetId,
+      destinationAddress ?? '',
+      DUMMY_UTXO_ID
+    )
+    const utxoSet = new utils.UtxoSet([utxos])
+    return readOnlySigner.exportP({
+      amount: amountInNAvax,
+      utxoSet,
+      destination: destinationChain,
+      toAddress: destinationAddress,
+      feeState
+    })
+  }
+
+  public async createDummyAddPermissionlessDelegatorTx({
+    amountInNAvax,
+    accountIndex,
+    avaxXPNetwork,
+    destinationAddress,
+    feeState
+  }: CreateExportPTxParams): Promise<UnsignedTx> {
+    const readOnlySigner = await this.getReadOnlyAvaSigner(
+      accountIndex,
+      avaxXPNetwork
+    )
+    const provider = await NetworkService.getAvalancheProviderXP(
+      !!avaxXPNetwork.isTestnet,
+      isDevnet(avaxXPNetwork)
+    )
+    const assetId = provider.getAvaxID()
+    const utxos = getStakeableOutUtxos(
+      amountInNAvax * 2n,
+      assetId,
+      destinationAddress ?? '',
+      DUMMY_UTXO_ID
+    )
+    const utxoSet = new utils.UtxoSet([utxos])
+    return readOnlySigner.addPermissionlessDelegator({
+      weight: amountInNAvax,
+      nodeId: 'NodeID-1',
+      subnetId: PChainId._11111111111111111111111111111111LPO_YY,
+      fromAddresses: [destinationAddress ?? ''],
+      rewardAddresses: [destinationAddress ?? ''],
+      start: BigInt(getUnixTime(new Date())),
+      // get the end date 1 month from now
+      end: BigInt(getUnixTime(new Date()) + 60 * 60 * 24 * 30),
```

Comment:
setting this end date here for this dummy tx is okay. since the end date does not add complexity for this tx, so it doesn't affect the txFee that is returned.